flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/codegen/CodeGenerator.scala

```diff
@@ -260,10 +260,14 @@ class CodeGenerator(
     val input1AccessExprs = for (i <- 0 until input1.getArity)
       yield generateInputAccess(input1, input1Term, i)
 
-    val input2AccessExprs = input2 match {
-      case Some(ti) => for (i <- 0 until ti.getArity)
-        yield generateInputAccess(ti, input2Term, i)
-      case None => Seq() // add nothing
+    val input2AccessExprs = if (input1.getArity < resultFieldNames.length) {
```

Comment:
Can this change be reverted if we use cogroup and an optional converter map? Or would you consider this as a bug fix?